tests/test_cloudpickle_wrapper.py

```diff
@@ -1,15 +1,37 @@
 import os
 import sys
 import pytest
+import io
 from tempfile import mkstemp
 from loky import set_loky_pickler
+from loky.backend.reduction import get_loky_pickler
 
 
 from .utils import check_subprocess_call
 
 
 class TestCloudpickleWrapper:
 
+    def test_isolated_pickler_dispatch_tables(self):
+        class A:
+            pass
+
+        class B:
+            pass
+
+        p1 = get_loky_pickler()(io.BytesIO(),
+                                reducers={A: lambda obj: (int, (42,))})
+        p2 = get_loky_pickler()(io.BytesIO(),
+                                reducers={B: lambda obj: (int, (42,))})
+
+        assert p1.dispatch_table is not p2.dispatch_table
```

Comment:
I think this assertion currently raises because `cloudpickle.Cloudpickler.dispatch_table` shadows `_pickle.Pickler.dispatch_table` when using dotted attribute-access syntax. This on the other hand
```python
assert pickle.Pickler.dispatch_table.__get__(p1) is not pickle.Pickler.dispatch_table.__get__(p2)
```
should pass. This comment holds for every following line in the test.

Reply:
Now that I also call `setattr`, it works as expected and this is not required. I also added some behavioral checks to make sure that the picklers are really using the custom reducers passed to the constructor.

Reply:
ok, sounds good.